src/remake.jl

```diff
@@ -19,14 +19,30 @@ function remake_buffer(sys, oldbuffer::AbstractArray, vals::Dict)
     if ArrayInterface.ismutable(oldbuffer) && !isa(oldbuffer, MArray)
         elT = Union{}
         for val in values(vals)
-            elT = Union{elT, typeof(val)}
+            elT = promote_type(elT, typeof(val))
         end
 
         newbuffer = similar(oldbuffer, elT)
-        setu(sys, keys(vals))(newbuffer, values(vals))
+        setu(sys, collect(keys(vals)))(newbuffer, elT.(values(vals)))
     else
         mutbuffer = remake_buffer(sys, collect(oldbuffer), vals)
         newbuffer = similar_type(oldbuffer, eltype(mutbuffer))(mutbuffer)
     end
     return newbuffer
 end
+
+mutable struct TupleRemakeWrapper
+    t::Tuple
```

Comment:
This won't infer, Tuple isn't concrete

Reply:
The type of the tuple can change, which is why it's not fully typed